ddl/index.go

```diff
@@ -966,112 +984,87 @@ func closeAddIndexWorkers(workers []*addIndexWorker) {
 	}
 }
 
-func (w *worker) waitTaskResults(workers []*addIndexWorker, taskCnt int, totalAddedCount *int64, startHandle int64) (int64, int64, error) {
-	var (
-		addedCount int64
-		nextHandle = startHandle
-		firstErr   error
-	)
-	for i := 0; i < taskCnt; i++ {
-		worker := workers[i]
-		result := <-worker.resultCh
-		if firstErr == nil && result.err != nil {
-			firstErr = result.err
-			// We should wait all working workers exits, any way.
-			continue
+func (w *worker) waitTaskFinish(cancel context.CancelFunc, reorgInfo *reorgInfo, availableWorkerCh chan *addIndexWorker, doingTaskCh chan *reorgIndexTask, errCh chan error) error {
```

Comment:
Add comments here, explain how why we need to wait the handling task finished sequentially